wpilibc/src/main/native/cpp/simulation/DifferentialDrivetrainSim.cpp

```diff
@@ -0,0 +1,91 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "frc/simulation/DifferentialDrivetrainSim.h"
+
+#include "frc/system/RungeKutta.h"
+
+using namespace frc;
+using namespace frc::sim;
+
+DifferentialDrivetrainSim::DifferentialDrivetrainSim(
+    LinearSystem<2, 2, 2>& plant, DifferentialDriveKinematics& kinematics,
+    DCMotor leftGearbox, double leftGearing, DCMotor rightGearbox,
+    double rightGearing)
+    : m_plant(plant),
+      m_rb(kinematics.trackWidth / 2.0),
+      m_leftMotor(leftGearbox),
+      m_rightMotor(rightGearbox),
+      m_leftGearing(leftGearing),
+      m_rightGearing(rightGearing) {}
+
+void DifferentialDrivetrainSim::SetInputs(units::volt_t leftVoltage,
+                                          units::volt_t rightVoltage) {
+  m_u << leftVoltage.to<double>(), rightVoltage.to<double>();
+}
+
+void DifferentialDrivetrainSim::Update(units::second_t dt) {
+  m_x = RungeKutta([this](auto& x, auto& u) { return Dynamics(x, u); }, m_x,
+                   m_u, dt);
+}
+
+double DifferentialDrivetrainSim::GetState(int state) { return m_x(state); }
+
+Vector<10> DifferentialDrivetrainSim::GetState() { return m_x; }
```

Comment:
Use `Eigen::Matrix<double, 10, 1>` instead of `Vector<>`. Vector is a custom typedef that doesn't exist in Eigen.